content/blog/2021/04/2021-04-16-deprecated-ruby-runtime.adoc

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title: "Deprecated ruby-runtime and associated plugins"
+tags:
+- ruby
+- plugins
+authors:
+- halkeye
+opengraph:
+  image: /images/post-images/2021-04-deprecating-ruby-runtime.png
+---
+Approximately
+https://github.com/jenkinsci/ruby-runtime-plugin/commit/d368b087fadb3282c9b537f0fa6d9a150b080c73[10 years ago, ruby-runtime was first released]. It was an experiment to let
+plugins to be written in ruby but still get integrated into the main
+java Jenkins process.
+
+Over the years though, the experiment has stopped being maintained, the
+dozen or so plugins that use it take a lot of user effort to make work,
+and the Jenkins ruby runtime does not support Java 11.
+Specifically the last few years have been really hard on the Jenkins
+core developers, as they still have to maintain all the hacks and
+processes to keep the ruby runtime barely working.
```

Comment:
This part and the impact do not align. Either we keep the hacks in, in which case this doesn't help, or we remove it, then there is "future releases won't support this anymore" impact.

Reply:
I like the idea that we should add a statement that support for those plugins will be removed in a future release.  The ruby runtime plugins already don't work with Java 11.  Let's make it clear that we're reducing the maintenance costs caused by the ruby runtime by declaring that a future Jenkins release will drop all support for ruby runtime.